lib/MongoDB/Op/_Aggregate.pm

```diff
@@ -133,11 +134,13 @@ sub execute {
     );
 
     my $op = MongoDB::Op::_Command->_new(
+        client          => $self->client,
         db_name         => $self->db_name,
         query           => Tie::IxHash->new(@command),
         query_flags     => {},
         bson_codec      => $self->bson_codec,
         ( $has_out ? () : ( read_preference => $self->read_preference ) ),
+        ( defined $self->session ? ( session => $self->session ) : () ),
```

Comment:
Shouldn't need this sort of conditionality.  As long as session is optional, then we can pass in `$self->session` (even if undef) and let the Command op (or the roles it consumes) deal with undef.